crates/cast/src/starknet_commands/account/add.rs

```diff
@@ -99,3 +122,119 @@ pub async fn add(
+            .ok_or_else(|| anyhow::anyhow!("No public_key in account JSON file"))?;
+        parse_number(pk)?
+    };
+    if public_key != private_key.verifying_key().scalar() {
+        bail!("Public key mismatch");
+    }
+
+    let address: FieldElement = if let Some(salt) = salt {
+        get_contract_address(salt, class_hash, &[public_key], FieldElement::ZERO)
+    } else {
+        let address = deployment
+            .get("address")
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("No address in account JSON file"))?;
+        parse_number(address)?
+    };
+
+    let account_json =
+        prepare_account_json(&private_key, address, deployed, Some(class_hash), salt);
+
+    Ok(account_json)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+
+    use crate::starknet_commands::account::add::import_from_keystore;
+    use camino::Utf8PathBuf;
+    use cast::helpers::constants::KEYSTORE_PASSWORD_ENV_VAR;
+    use serde_json::json;
+
+    #[test]
+    fn test_import_from_keystore() {
+        let keystore_path = Utf8PathBuf::from("tests/data/keystore/my_key.json");
+        let account_path = Utf8PathBuf::from("tests/data/keystore/my_account.json");
+
+        env::set_var(KEYSTORE_PASSWORD_ENV_VAR, "123");
+        let account_json = import_from_keystore(&keystore_path, &account_path).unwrap();
+
+        assert_eq!(
+            account_json,
+            json!(
+                {
+                    "address": "0xcce3217e4aea0ab738b55446b1b378750edfca617db549fda1ede28435206c",
+                    "class_hash": "0x646a72e2aab2fca75d713fbe4a58f2d12cbd64105621b89dc9ce7045b5bf02b",
+                    "deployed": true,
+                    "private_key": "0x55ae34c86281fbd19292c7e3bfdfceb4",
+                    "public_key": "0xe2d3d7080bfc665e0060a06e8e95c3db3ff78a1fec4cc81ddc87e49a12e0a",
+                }
+            )
+        );
+    }
+
+    #[test]
+    fn test_import_from_keystore_undeployed() {
+        let keystore_path = Utf8PathBuf::from("tests/data/keystore/my_key.json");
+        let account_path = Utf8PathBuf::from("tests/data/keystore/my_account_undeployed.json");
+
+        env::set_var(KEYSTORE_PASSWORD_ENV_VAR, "123");
+        let account_json = import_from_keystore(&keystore_path, &account_path).unwrap();
+
+        assert_eq!(
+            account_json,
+            json!(
+                {
+                    "address": "0xcce3217e4aea0ab738b55446b1b378750edfca617db549fda1ede28435206c",
+                    "class_hash": "0x646a72e2aab2fca75d713fbe4a58f2d12cbd64105621b89dc9ce7045b5bf02b",
+                    "deployed": false,
+                    "private_key": "0x55ae34c86281fbd19292c7e3bfdfceb4",
+                    "public_key": "0xe2d3d7080bfc665e0060a06e8e95c3db3ff78a1fec4cc81ddc87e49a12e0a",
+                    "salt": "0x14df438ac6825165c7a0af29decd5892528b763a333f93a5f6b12980dbddd9f",
+                }
+            )
+        );
+    }
+}
```

Comment:
it would be cool to have more tests. I know the description says they will be added later, just want to make sure there will be some more :+1: